File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.cpp
class Solution {
public:
    int pivotInteger(int n) {
        unordered_map<int, int> mp;
        int prefixSum = 0;
        for (int num = 1; num <= n; num++) {
            prefixSum += num;
            mp[num] = prefixSum;
        }
        prefixSum = 0;
        for (int num = n; num > 0; num--) {
            prefixSum += num;
            if (mp[num] == prefixSum) return num;
        }
        return -1;
    }
};

File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.rs
impl Solution {
    pub fn pivot_integer(n: i32) -> i32 {
        let sum = n * (n + 1) / 2;
        let x = (sum as f64).sqrt() as i32;
        if x * x == sum {
            x
        } else {
            -1
        }
    }
}